samcli/local/events/api_event.py

```diff
@@ -205,7 +208,9 @@ def __init__(
         if not isinstance(stage_variables, dict) and stage_variables is not None:
             raise TypeError("'stage_variables' must be of type dict or None")
 
-        self.version = "1.0"
+        # v1 payloads and rest api payloads are identical save for the version field
+        if api_type == Route.HTTP:
+            self.version = "1.0"
```

Comment:
It looks like storing `self.version` is redundant as it can only have constant value which is not used by the change below as well. Can you check if it has other usages from other places, if not we may remove it completely?